### PR TITLE
Feature/vote 879 state page styling

### PR DESCRIPTION
**Pull request by SamiraMSadat**
## Jira ticket (required)
https://bixal-projects.atlassian.net/browse/VOTE-897

## Description
Styling of all state page and it's components

### Post-deploy steps
1. Run lando retune
2. Run npm run build in the theme folder

### Testing steps
1. Create a state page in the admin interface with at least two NVRF links for the selector
2. Verify the selector works as intended in the browser
3. Verify that the pages are styled correctly
4. Change the registration types in the admin interface and observe that they are shown/hidden on the page as the data changes

### CODE REVIEW

File: web/themes/custom/vote_gov/templates/field/field--block-content--field-heading--registration-form-selector.html.twig
Comment (SamiraMSadat) [inline web/themes/custom/vote_gov/templates/field/field--block-content--field-heading--registration-form-selector.html.twig:31]: The @state_name is not replaced within the string coming from item.content. How should we handle this?

File: web/themes/custom/vote_gov/templates/partial/registration-date-online.html.twig
Comment (SamiraMSadat) [inline web/themes/custom/vote_gov/templates/partial/registration-date-online.html.twig:10]: The online deadline moves to another line because that is how the Double Field module is styled. We need to style this div with inline:block, but I am not sure where it is best to do this since the styles are coming from a module.

File: web/themes/custom/vote_gov/templates/partial/registration-date-in-person.html.twig
Comment (SamiraMSadat) [inline web/themes/custom/vote_gov/templates/partial/registration-date-in-person.html.twig:10]: See online deadline comment, same issue

File: web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig:14]: We need to wrap non-required field output in a condition so that when the value is missing the content will not display.
An example:
```
{% if content.field_registration_link | render %}
// Show the content
{% endif %}
```

File: web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig:18]: The override link value should replace the standard value if it is present.

File: web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig:20]: Same comment as the other: We need a conditional wrapping non-required values and the override link value some replace the standard link value when present.

File: web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig:36]: Same comment as the other: We need a conditional wrapping non-required values and the override link value some replace the standard link value when present.

File: web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig:42]: last updated would be the revision date for the node
`{{ node.revision_timestamp.value | date('F d, Y') }}`

File: web/themes/custom/vote_gov/templates/partial/registration-date-online.html.twig
Comment (rayestrada) [reply web/themes/custom/vote_gov/templates/partial/registration-date-online.html.twig:10]: This is how I would handle this template. This uses the default filter to provide a fallback. Then you can check if there is a deadline before you print the content. Then it uses the striptags filter to remove extra markup coming from doublefield.
```
{% set deadline = content.field_online_deadline | field_value | default(content.field_g_online_deadline | field_value) %}

{% if deadline %}
  <li><span class="bold-text">{{ 'Online registration deadline: ' | t }}</span>{{ deadline | render | striptags }}</li>
{% endif %}
```

File: web/themes/custom/vote_gov/templates/partial/registration-date-in-person.html.twig
Comment (rayestrada) [reply web/themes/custom/vote_gov/templates/partial/registration-date-in-person.html.twig:10]: See my comment on the online template.

File: web/themes/custom/vote_gov/templates/partial/registration-date-mail.html.twig
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/registration-date-mail.html.twig:9]: See the online template comment. It might make sense to use the same method for this file.

**Review by rayestrada — CHANGES_REQUESTED** [review]

File: web/themes/custom/vote_gov/php-includes/preprocess.inc
Comment (SamiraMSadat) [reply web/themes/custom/vote_gov/php-includes/preprocess.inc:42]: I added the conditional logic to the title template so that only the state pages are affected.

File: web/themes/custom/vote_gov/templates/block/block-content--registration-form-selector--full.html.twig
Comment (SamiraMSadat) [reply web/themes/custom/vote_gov/templates/block/block-content--registration-form-selector--full.html.twig:20]: I created the new ticket.

**Review by rayestrada — DISMISSED** [review]

**Comment by rayestrada** [comment]
Waiting for approval and merge of #83 before merging,

**Review by rayestrada — APPROVED** [review]

**Pull request merged**
